src/main/scala/is/hail/methods/LDPrune.scala

```diff
@@ -303,7 +303,7 @@ object LDPrune {
       if (partitionId == partitionIndices(0))
         Array(partitionId)
       else {
-        val startLocus = rangeBounds(partitionId - 1) // this is the best estimate of the first locus of the partition
+        val startLocus = rangeBounds(partitionId).start.asInstanceOf[UnsafeRow].getAs[Locus](0)
```

Comment:
I think this code can now be generic: don't check partitionId == 0, just do the else case using the partition start.  @jigold can you confirm?f

Reply:
Yes. That should be fine.